extensions/amp-youtube/1.0/youtube.js

```diff
@@ -0,0 +1,227 @@
+  credentials,
+  style,
+  ...rest
+}) {
+  const datasourceExists =
+    !(videoid && liveChannelid) && (videoid || liveChannelid);
+
+  if (!datasourceExists) {
+    throw new Error(
+      'Exactly one of data-videoid or data-live-channelid should be present for <amp-youtube>'
+    );
+  }
+
+  let src = getEmbedUrl(credentials, videoid, liveChannelid);
+  if (!('playsinline' in params)) {
+    params['playsinline'] = '1';
+  }
+  if ('autoplay' in params) {
+    delete params['autoplay'];
+    autoplay = true;
+  }
+
+  if (autoplay) {
+    if (!('iv_load_policy' in params)) {
+      params['iv_load_policy'] = `${PlayerFlags.HIDE_ANNOTATION}`;
+    }
+
+    // Inline play must be set for autoplay regardless of original value.
+    params['playsinline'] = '1';
+  }
+
+  if ('loop' in params) {
+    loop = true;
+    delete params['loop'];
+  }
+
+  if (loop) {
+    if ('playlist' in params) {
+      params['loop'] = '1';
+    } else if ('loop' in params) {
+      delete params['loop'];
+    }
+  }
+
+  src = addParamsToUrl(src, params);
+
+  const onMessage = ({data, currentTarget}) => {
+    data = objOrParseJson(data);
+    if (data == null) {
+      return;
+    }
+    if (data.event == 'initialDelivery') {
+      dispatchCustomEvent(currentTarget, VideoEvents.LOADEDMETADATA);
+      return;
+    }
+    const {info} = data;
+    if (info == undefined) {
+      return;
+    }
+    const playerState = info['playerState'];
+    if (data.event == 'infoDelivery' && playerState == 0 && loop) {
+      currentTarget.contentWindow./*OK*/ postMessage(
+        JSON.stringify(
+          dict({
+            'event': 'command',
+            'func': 'playVideo',
+          })
+        ),
+        '*'
+      );
+    }
+    if (data.event == 'infoDelivery' && playerState != undefined) {
+      dispatchCustomEvent(currentTarget, PlayerStates[playerState.toString()]);
+    }
+    if (data.event == 'infoDelivery' && info['muted']) {
+      dispatchCustomEvent(currentTarget, mutedOrUnmutedEvent(info['muted']));
+      return;
+    }
+  };
+
+  const sendMessage = (method) => {
+    return JSON.stringify(
+      dict({
+        'event': 'command',
+        'func': methods[method],
+      })
+    );
+  };
+
+  return (
+    <VideoWrapper
+      component={VideoIframe}
+      autoplay={autoplay}
+      src={src}
+      style={style}
+      size={true}
+      layout={true}
+      paint={true}
+      onMessage={onMessage}
+      sendMessage={sendMessage}
+      onLoad={(event) => {
+        const {currentTarget} = event;
+        dispatchCustomEvent(currentTarget, 'canplay');
+        currentTarget.contentWindow./*OK*/ postMessage(
+          JSON.stringify(
+            dict({
+              'event': 'listening',
+            })
+          ),
+          '*'
+        );
+        dispatchCustomEvent(currentTarget, VideoEvents.LOAD);
+      }}
+      sandbox="allow-scripts allow-same-origin allow-presentation"
+      {...rest}
+    ></VideoWrapper>
+  );
```

Comment:
- `sendMessage` has been renamed. Please update your branch.
- `VideoIframe` already uses a `VideoWrapper`, so you don't need the additional nesting.
- `layout`/`size`/`paint` should be handled by the wrapper.

```suggestion
  return (
    <VideoIframe
      autoplay={autoplay}
      src={src}
      style={style}
      onMessage={onMessage}
      makeMethodMessage={makeMethodMessage}
      onLoad={(event) => {
        const {currentTarget} = event;
        dispatchCustomEvent(currentTarget, 'canplay');
        currentTarget.contentWindow./*OK*/ postMessage(
          JSON.stringify(
            dict({
              'event': 'listening',
            })
          ),
          '*'
        );
        dispatchCustomEvent(currentTarget, VideoEvents.LOAD);
      }}
      sandbox="allow-scripts allow-same-origin allow-presentation"
      {...rest}
    ></VideoWrapper>
  );
```